rust/repositories.bzl

```diff
@@ -392,7 +418,14 @@ def _rust_toolchain_repository_impl(ctx):
     if ctx.attr.rustfmt_version:
         BUILD_components.append(_load_rustfmt(ctx))
 
-    for target_triple in [ctx.attr.exec_triple] + ctx.attr.extra_target_triples:
+    for target_triple in [ctx.attr.exec_triple]:
+        BUILD_components.append(_load_rust_stdlib(ctx, target_triple))
+        if ctx.attr.dev_components:
+            # No BUILD file is needed because components will match the stdlib
+            # BUILD file globs
+            _load_rustc_dev_nightly(ctx, target_triple)
+
+    for target_triple in ctx.attr.extra_target_triples:
         BUILD_components.append(_load_rust_stdlib(ctx, target_triple))
```

Comment:
`ctx.attr.extra_target_triples` contains the wasm target. While there is a stdlib package, there are no rustc-dev components shipped for this target. Because of this, I split this into two different loops. This means the `if` statement in the first loop only has to check if dev_components is True. Another option might be to keep one loop, and have the if statement be:
```python
if ctx.attr.dev_components and target_triple not in ctx.attr.extra_target_triples:
```
I prefer the two loops personally, but I can also switch to the longer `if` statement if there is a strong preference for it by the maintainers.

Reply:
After reading the code, I think one loop will make reading the code easier, and adding a comment on why excluding extra_target_triples make sense (e.g. "it contains wasm and the likes, which does not have a dev component")